File: general-server/src/main/java/com/song/middleware/server/storage/ServiceStorageComponent.java
package com.song.middleware.server.storage;

import com.song.middleware.server.domain.ServiceDO;
import java.util.List;
import java.util.concurrent.Future;

/**
 * Created by song on 2017/8/5.
 */
public interface ServiceStorageComponent {

    ServiceDO syncQuery(String name, String ip, String port);

    List<ServiceDO> syncQuery(String name);

    Future<List<ServiceDO>> asyncQuery(String name);

    void syncRegister(ServiceDO serviceDO);

    Future<?> asyncRegister(ServiceDO serviceDO);

    void syncUnregister(String name, String ip, String port);
}

File: general-server/src/main/java/com/song/middleware/server/storage/ServiceStorageComponent.kt
package com.song.middleware.server.storage

import com.song.middleware.server.domain.ServiceDO
import java.util.concurrent.Future

/**
 * Created by song on 2017/8/5.
 */
interface ServiceStorageComponent {

    fun syncQuery(name: String, ip: String, port: String): ServiceDO

    fun syncQuery(name: String): List<ServiceDO>

    fun asyncQuery(name: String): Future<List<ServiceDO>>

    fun syncRegister(serviceDO: ServiceDO)

    fun asyncRegister(serviceDO: ServiceDO): Future<*>

    fun syncUnregister(name: String, ip: String, port: String)
}
